syntax = "proto3";

service CurrentTimestamp {
  rpc getCurrentTime(TimeRequest) returns (TimeResponse) {}
  rpc sayHello(HelloRequest) returns (HelloResponse) {}
}

service GuessCountry {
  rpc geo(stream GeoRequest) returns (stream GeoResponse) {}
}

message GeoRequest {
  float longitude = 1;
  float latitude = 2;
}

message GeoResponse {
  string country_name = 1;
}

message HelloRequest {
  optional string name = 1;
}

message HelloResponse {
  string message = 1;
}

message TimeResponse {
  string current_time = 1;
  optional string tz = 2;
}

message TimeRequest {
  optional string timezone = 1;
}